### PR TITLE
FLAG-70: Get flagged patient based on tags is not working

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

Fixed the error in get flagged patients based on tags

[Screencast from 07-12-2024 10:15:35 PM.webm](https://github.com/user-attachments/assets/157b7443-9397-4851-915f-56814e4ec703)

## Issue I worked on

see https://openmrs.atlassian.net/browse/FLAG-70

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.
  
  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by wikumChamith** [comment]
@ManojLL doesn't this work on the master?

**Comment by ManojLL** [comment]
> @ManojLL doesn't this work on the master?

No, get an error on master

image

File: omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java [hunk 1]
```diff
@@ -161,7 +161,21 @@ public ModelAndView processSubmit(@ModelAttribute("filter") Filter filter, Bindi
 		ModelMap model = new ModelMap();
 		model.addAttribute("flaggedPatients", flaggedPatients.getMemberIds());
```
Comment (wikumChamith) [inline omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java:162]: Do we need this now?

**Review by wikumChamith — COMMENTED** [review]
Can't we fix this by changing the frontend code in the jsp file?

File: omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java (hunk 1 above)
Comment (ManojLL) [reply omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java]: I removed it

**Comment by ManojLL** [comment]
> Can't we fix this by changing the frontend code in the jsp file?

 The issue was in the controller it was not map the patient list as it required in jsp. we are using same jsp for get flagged patients for flags and get flagged patient for tags. I do not think we can fix it from jsp front end side.

File: omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java [hunk 2]
```diff
@@ -159,9 +159,21 @@ public ModelAndView processSubmit(@ModelAttribute("filter") Filter filter, Bindi
 		
 		// create the model map
 		ModelMap model = new ModelMap();
-		model.addAttribute("flaggedPatients", flaggedPatients.getMemberIds());
 		model.addAttribute("allPatients", allPatients);
-		
+		List<Map<String, Object>> fpl = new ArrayList<>();
```
Comment (wikumChamith) [inline omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java:163]: Can we use better variable names?

Comment (ManojLL) [reply omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java]: rename the variable names

File: omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java [hunk 3]
```diff
@@ -159,9 +159,21 @@ public ModelAndView processSubmit(@ModelAttribute("filter") Filter filter, Bindi
 		
 		// create the model map
 		ModelMap model = new ModelMap();
-		model.addAttribute("flaggedPatients", flaggedPatients.getMemberIds());
 		model.addAttribute("allPatients", allPatients);
-		
+		List<Map<String, Object>> flaggedPatientList = new ArrayList<>();
```
Comment (wikumChamith) [inline omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java:163]: Can't we use Map<String, Integer> here?

Comment (ManojLL) [reply omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java]: yes we can. I updated map value type object to Integer.

File: omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java [hunk 4]
```diff
@@ -159,9 +159,21 @@ public ModelAndView processSubmit(@ModelAttribute("filter") Filter filter, Bindi
 		
 		// create the model map
 		ModelMap model = new ModelMap();
-		model.addAttribute("flaggedPatients", flaggedPatients.getMemberIds());
 		model.addAttribute("allPatients", allPatients);
-		
+		List<Map<String, Integer>> flaggedPatientList = new ArrayList<>();
+
+		Set<Integer> flaggedPatientIds = flaggedPatients.getMemberIds();
+		for (Integer patientId : flaggedPatientIds) {
```
Comment (wikumChamith) [inline omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java:166]: I think we could simplify this code using streams. Can you give it a try?

Comment (ManojLL) [reply omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java]: I refactor the code using stream API

**Pull request merged**